libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/VoiceInstructionsDownloadTrigger.kt

```diff
@@ -0,0 +1,149 @@
+package com.mapbox.navigation.ui.voice.api
+
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesExtra
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.directions.session.RoutesUpdatedResult
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.utils.internal.Time
+import com.mapbox.navigation.utils.internal.ifNonNull
+
+/**
+ * Class that is responsible for listening to relevant updates and triggering
+ * voice instructions predownloading when needed.
+ * Register and unregister with [MapboxNavigation.registerVoiceInstructionsTriggerObserver] and
+ * [MapboxNavigation.unregisterVoiceInstructionsTriggerObserver].
+ */
+class VoiceInstructionsDownloadTrigger internal constructor(
```

Comment:
Can we start by making this experimental? in case there needs to be iteration api